nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp

```diff
@@ -175,16 +191,16 @@ geometry_msgs::msg::TwistStamped RegulatedPurePursuitController::computeVelocity
 
   double linear_vel, angular_vel;
 
-  // Find distance^2 to look ahead point (carrot) in robot base frame
-  // This is the chord length of the circle
-  const double carrot_dist2 =
-    (carrot_pose.pose.position.x * carrot_pose.pose.position.x) +
-    (carrot_pose.pose.position.y * carrot_pose.pose.position.y);
+  double desired_curvature = calculateCurvature(carrot_pose.pose.position);
 
-  // Find curvature of circle (k = 1 / R)
-  double curvature = 0.0;
-  if (carrot_dist2 > 0.001) {
-    curvature = 2.0 * carrot_pose.pose.position.y / carrot_dist2;
+  double regulation_curvature;
+  if (params_->use_fixed_curvature_lookahead) {
+    auto curvature_lookahead_pose = getLookAheadPoint(
+      params_->curvature_lookahead_dist,
+      transformed_plan);
+    regulation_curvature = calculateCurvature(curvature_lookahead_pose.pose.position);
```

Comment:
if you initialize `regulation_curvature` with `curvature`, then you can remove the `else` statement.